ledger/acctupdates.go

```diff
@@ -640,63 +526,42 @@ func (au *accountUpdates) committedUpTo(committedRound basics.Round) (retRound b
 // operation to a syncer goroutine. The one caveat is that when storing a catchpoint round, we would want to
 // wait until the catchpoint creation is done, so that the persistence of the catchpoint file would have an
 // uninterrupted view of the balances at a given point of time.
-func (au *accountUpdates) produceCommittingTask(committedRound basics.Round, dbRound basics.Round) (dc deferredCommit) {
-	var isCatchpointRound, hasMultipleIntermediateCatchpoint bool
+func (au *accountUpdates) produceCommittingTask(committedRound basics.Round, dbRound basics.Round, dcc *deferredCommitContext) *deferredCommitContext {
+	var isCatchpointRound bool
 	var offset uint64
 	au.accountsMu.RLock()
 	defer au.accountsMu.RUnlock()
 
 	var pendingDeltas int
-	lookback := basics.Round(config.Consensus[au.versions[len(au.versions)-1]].MaxBalLookback)
-	if committedRound < lookback {
-		return
+
+	if committedRound < dcc.lookback {
+		return nil
 	}
 
-	newBase := committedRound - lookback
+	newBase := committedRound - dcc.lookback
 	if newBase <= dbRound {
 		// Already forgotten
-		return
+		return nil
 	}
 
 	if newBase > dbRound+basics.Round(len(au.deltas)) {
-		au.log.Panicf("produceCommittingTask: block %d too far in the future, lookback %d, dbRound %d (cached %d), deltas %d", committedRound, lookback, dbRound, au.cachedDBRound, len(au.deltas))
+		au.log.Panicf("produceCommittingTask: block %d too far in the future, lookback %d, dbRound %d (cached %d), deltas %d", committedRound, dcc.lookback, dbRound, au.cachedDBRound, len(au.deltas))
 	}
 
-	hasIntermediateCatchpoint := false
-	hasMultipleIntermediateCatchpoint = false
 	// check if there was a catchpoint between au.dbRound+lookback and newBase+lookback
 	if au.catchpointInterval > 0 {
-		nextCatchpointRound := ((uint64(dbRound+lookback) + au.catchpointInterval) / au.catchpointInterval) * au.catchpointInterval
+		nextCatchpointRound := ((uint64(dbRound+dcc.lookback) + au.catchpointInterval) / au.catchpointInterval) * au.catchpointInterval
```

Comment:
This logic has been copied to catchpointTracker.produceCommittingTask but is still needed here, could they share?